code/common/checkinputs.q

```diff
@@ -0,0 +1,257 @@
+        '`$.checkinputs.formatstring[.schema.errors[`checkrepeatparams;`errormessage];where repeats]];
+    :dict;};
+
+// loop thorugh input parameters to execute parameter specific checks
+checkeachparam:{[dict;isrequired]
+    config:select from .checkinputs.checkinputsconfig where parameter in key dict,required=isrequired;
+    :checkparam/[dict;config];
+  };
+
+// extract parameter specific function from config to check the input
+checkparam:{[dict;config]
+    (first config[`checkfunction])[dict;first config`parameter]};
+
+// check tablename parameter is of type symbol
+checktable:{[dict;parameter]:checktype[-11h;dict;parameter];};
+
+// check that endtime is of temporal type and that it is greater than or equal to starttime
+checkendtime:{[dict;parameter]
+    dict:checktimetype[dict;parameter];
+    :checktimeorder dict};
+
+// check that inputted value is of valid type: -12 -14 -15h
+checktimetype:{[dict;parameter]:checktype[-12 -14 -15h;dict;parameter];};
+
+// check timecolumn is of type symbol
+// check starttime <= endtime
+checktimecolumn:{[dict;parameter]:checktype[-11h;dict;parameter];};
+
+// check starttime <= endtime
+checktimeorder:{[dict]
+    if[dict[`starttime] > dict`endtime;'`$.schema.errors[`checktimeorder;`errormessage]];
+    :dict;};
+
+// check instruments, columns and grouping are of type symbol
+checksyminput:{[dict;parameter]
+    :checktype[-11 11h;dict;parameter];};
+
+// check aggregations are of type dictionary, that the dictionary has symbol keys, that 
+// the dictionary has symbol values
+checkaggregations:{[dict;parameter]
+    dict:checktype[99h;dict;parameter];
+    input:dict parameter;
+    if[not 11h~abs type key input;
+        '`$.schema.errors[`checkaggregationkey;`errormessage],.schema.examples[`aggregations1;`example]];
+    if[not all 11h~/:abs raze type''[get input];
+        '`$.schema.errors[`checkaggregationparameter;`errormessage],.schema.examples[`aggregations1;`example]];
+    :dict;
+  };
+
+// check that timebar parameter has three elements of respective types: numeric, symbol,
+// symbol.
+checktimebar:{[dict;parameter]
+    input:dict parameter;
+    if[not(3=count input)&0h~type input;
+        '`$.schema.errors[`timebarlength;`errormessage]];
+    input:(`$"timebar first arguement";`$"timebar second arguement";`$"timebar third arguement")!input;
+    .checkinputs.checktype[-6 -7h;input;`$"timebar first arguement"];
+    .checkinputs.checktype[-11h;input;`$"timebar second arguement"];
+    .checkinputs.checktype[-11h;input;`$"timebar third arguement"];
+    :dict;
+  };
+
+// check that filters parameter is of type dictionary, has symbol keys, the values are 
+// in (where function;value(s)) pairs and (not)within filtering functions have two values
+// associated with it.
+checkfilters:{[dict;parameter]
+    dict:checktype[99h;dict;parameter];
+    (dict parameter):@[(dict parameter);where {not all 0h=type each x}each (dict parameter);enlist];
```

Comment:
Convention is for square brackets:
```suggestion
    dict[parameter]:@[(dict parameter);where {not all 0h=type each x}each (dict parameter);enlist];
```
Also wondering if you can do something like this
```
dict:@[dict;parameter;{$[not all 0h=type each x;enlist;]x}];
```